### PR TITLE
[feat] Allow custom HTTP headers with the `httpjson` log handler

### LINKED ISSUE #2850: HTTPJSONHandler http POST customisable header
https://github.com/reframe-hpc/reframe/blob/088207c72a95d10631a260f5dcdf76490f984216/reframe/core/logging.py#L605-L609

It would be nice to be able to customise the header of the http POST request via config file. 
This may be useful for some services requiring  auth via API Tokens while using the `httpjson` handler.

### PR DESCRIPTION
**Pull request by Lewih**
I tested this with ncat and a small config file as 

```python
site_configuration = {
    'logging': [
        {
        'handlers_perflog': [

            {
            'type': 'httpjson',
            'url': 'http://localhost:12345',
            'level': 'info',
            'extras': {
                'facility': 'reframe',
                'data-version': '1.0'
            },
            'debug': False,
            'extra_headers': {
               'ApiKey': 'test'
            }
            }
            ]
        }
    ]
}
```
Closes #2850.

~~Should we implement a test ?~~

### CODE REVIEW

**Review by teojgo — CHANGES_REQUESTED** [review]
You would need to add this to the configuration schema as well.

**Comment by Lewih** [comment]
> You would need to add this to the configuration schema as well.

gonna add it as ```"extra_headers": {"type": "object"},```

**Comment by Lewih** [comment]
Documentation should be updated as well

**Comment by Lewih** [comment]
@teojgo , The PR now follows the community guidelines and the schema has been updated.

As far as I have seen, no test is included for `extras` and the others entries of `httpjson` handler. I would postpone a more complete test implementation for the handler for another PR.

**Comment by teojgo** [comment]
> @teojgo , The PR now follows the community guidelines and the schema has been updated.
> 
> As far as I have seen, no test is included for `extras` and the others entries of `httpjson` handler. I would postpone a more complete test implementation for the handler in another PR.

@Lewih Thanks for the PR and the update. Yes, we don't have any testing for the handler at the moment but suggestions are always welcome.

**Review by teojgo — APPROVED** [review]

**Comment by Lewih** [comment]
> @Lewih Thanks for the PR and the update. Yes, we don't have any testing for the handler at the moment but suggestions are always welcome.

I opened an issue #2858 in order to track this.

**Comment by Lewih** [comment]
Synched with upstream branch in order to re-run the test to make sure nothing breaks, then PR should be able to merge.

File: docs/config_reference.rst [hunk 1]
```diff
@@ -1345,6 +1345,15 @@ The additional properties for the ``httpjson`` handler are the following:
...
    These keys will be excluded from the log record that will be sent to the server.
 
+.. py:attribute:: logging.handlers..httpjson..extra_headers
+
+.. py:attribute:: logging.handlers_perflog..httpjson..extra_headers
+
+   :required: No
+   :default: ``{}``
+
+   A set of optional key/value pairs to be sent as HTTP message headers(e.g. API keys).
```
Comment (vkarak) [inline docs/config_reference.rst]:
```suggestion
   A set of optional key/value pairs to be sent as HTTP message headers (e.g. API keys).
```

File: reframe/core/logging.py [hunk 2]
```diff
@@ -514,7 +515,7 @@ def _create_httpjson_handler(site_config, config_prefix):
         getlogger().warning('httpjson: running in debug mode; '
                             'no data will be sent to the server')
 
-    return HTTPJSONHandler(url, extras, ignore_keys, json_formatter, debug)
+    return HTTPJSONHandler(url, extras, ignore_keys, json_formatter, extra_headers, debug)
```
Comment (vkarak) [inline reframe/core/logging.py]:
```suggestion
    return HTTPJSONHandler(url, extras, ignore_keys, json_formatter,
                           extra_headers, debug)
```

**Review by vkarak — APPROVED** [review]
Lgtm; I just did a couple of minor style enhancements.

**Comment by vkarak** [comment]
We'll merge this after #2861 is addressed, which has broken our CI.

**Pull request merged**